src/components/PopupWithForm.js

```diff
@@ -1,33 +1,40 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, submitButton) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupElement.querySelector(".modal__submit-button");
+    // this._originalButtonText = this._submitButton.textContent;
   }
 
   _getInputValues() {
-    //collects data from all the input fields
-    this._inputList = this._popupElement.querySelectorAll(".modal__input");
-    this._values = {};
-    this._inputList.forEach((input) => {
-      this._values[input.name] = input.value;
-    });
-
-    return this._values;
+    const inputList = this._popupElement.querySelectorAll(".modal__input");
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`